### PR TITLE
Add explicit parsing rules for FETCH integration

### LINKED ISSUE #42: Evaluate current parsing error strategy
During code review [here](https://chromium-review.googlesource.com/c/chromium/src/+/741465), Mike West said: 

> Right now, you're setting `duration_` to zero if someone passes in `duration=whatever`. Did you consider something a little more draconian? Reject the header entirely, perhaps, with an appropriate console warning for non-numeric values?

I responded: 

> At present, this header resolves to a duration of 0.0:
> name; duration=foo; duration=123
> 
> I think that holds up, spec-wise, so my preference would be to leave as-is.
> -first specified parameter "wins" (others are ignored)
> (https://w3c.github.io/server-timing/#the-server-timing-header-field)
> -failure to parse a duration as a double should result in 0.0. 
> https://w3c.github.io/server-timing/#-dfn-server-timing-header-parsing-algorithm-dfn-

It is worth having a discussion about how draconian we should be with regard to parse errors.

### LINKED ISSUE #55: Fetch integration is missing
It's unclear when this header is processed.

### LINKED ISSUE #69: Creation time of PerformanceServerTiming objects unclear
When exactly is this object created in relation to the event loop? In particular the trailer requirements seem inadequate, but everything else is quite vague as well.

### PR DESCRIPTION
**Pull request by noamr**
Depends on https://github.com/whatwg/fetch/pull/1406 
Closes #42 
Closes #55
Closes #69

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/server-timing/pull/87.html" title="Last updated on Apr 21, 2022, 12:20 PM UTC (e275d9f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/server-timing/87/bc412c6...e275d9f.html" title="Last updated on Apr 21, 2022, 12:20 PM UTC (e275d9f)">Diff</a>

### CODE REVIEW

**Comment by noamr** [comment]
> ### 💥 Error: 500 Internal Server Error 💥
> [PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Feb 28, 2022, 8:36 AM UTC)_.
> 
> More

Build succeeded, not sure why it still says "failed to build"

File: index.html [hunk 1]
```diff
@@ -107,6 +108,53 @@
   - The server and/or any relevant intermediaries are in full control of which metrics are communicated to the user agent and when. For example, access to some metrics may be restricted due to privacy or security reasons - see <a href="#privacy-and-security"></a> section.
 </div>
 
+To <dfn>parse a server-timing header field</dfn> given a string |field|:
+
+1. Let |position| be a [=string/position variable=], initially pointing at the start of |field|.
+
+1. Let |rawName| be the result of [=collecting a sequence of code points=] that are not equal to U+003B (;), given |position|.
+
+1. Let |name| be the result of [=strip leading and trailing ASCII whitespace|stripping=] |rawName|.
```
Comment (yoavweiss) [inline index.html:117]: The algorithm is defined as operating in place rather than returning a result.

**Review by yoavweiss — APPROVED** [review]
LGTM % nits

It might've been nicer if we could fit some of it into a [structured field](https://datatracker.ietf.org/doc/html/rfc8941) processing model, but it doesn't seem to fit.

**Comment by noamr** [comment]
> LGTM % nits
> 
> It might've been nicer if we could fit some of it into a [structured field](https://datatracker.ietf.org/doc/html/rfc8941) processing model, but it doesn't seem to fit.

It would have been nice, but it doesn't work with a combination of tokens and quoted strings.

File: index.html [hunk 2]
```diff
@@ -107,6 +108,53 @@
   - The server and/or any relevant intermediaries are in full control of which metrics are communicated to the user agent and when. For example, access to some metrics may be restricted due to privacy or security reasons - see <a href="#privacy-and-security"></a> section.
 </div>
 
+To <dfn>parse a server-timing header field</dfn> given a string |field|:
+
+1. Let |position| be a [=string/position variable=], initially pointing at the start of |field|.
+
+1. Let |rawName| be the result of [=collecting a sequence of code points=] that are not equal to U+003B (;), given |position|.
```
[the remark this answers is not in the document]
Comment (noamr) [reply index.html:115]: I tried to be consistent with FETCH, everything there uses this style:
e.g. https://fetch.spec.whatwg.org/#simple-range-header-value

[the remark this answers is not in the document]
Comment (yoavweiss) [reply index.html]: OK, just add a "from `field`" then, to clarify which string this is reading from.

File: index.html (hunk 1 above)
Comment (noamr) [reply index.html]: Fixed

**Comment by noamr** [comment]
Will merge once the corresponding FETCH PR is merge, with a ref to the new struct item.

File: index.html [hunk 3]
```diff
@@ -145,51 +203,25 @@
...
 
-<p>For each resource [= fetch | fetched =] by the current [=browsing context=], excluding resources fetched by cross-origin stylesheets fetched with `no-cors` policy, set the {{PerformanceResourceTiming/serverTiming}} attribute on the <a data-cite="RESOURCE-TIMING-2#step-create-object">newly created</a> {{PerformanceResourceTiming}} object to:
-  <ol>
-    <li>An empty [=sequence=], if the return value from the "timing allow check" algorithm (as defined in [[RESOURCE-TIMING-2]]) is `fail`.</li>
-  <li>An empty [=sequence=], if the user agent chooses to limit their support to <span data-cite="html">[=secure contexts=]</span> and the <a data-cite="resource-timing-2#dfn-current-document">current document</a>'s [=environment settings object=] is <a data-cite="html" data-lt="non-secure contexts">not contextually secure</a>.</li>
-    <li>The return value of the [=server-timing header parsing algorithm=], otherwise.</li>
-  </ol>
-</p>
+  1. [=list/iterate|For each=] |field| in <a>this</a>'s <a data-cite="RESOURCE-TIMING#dfn-timing-info">timing info</a>'s [ =fetch timing info/server timing headers =]: <!-- TODO: fix link after FETCH PR is merged-->
```
Comment (annevk) [inline index.html:212]: I added a hyphen to this field. It's now "server-timing headers".

**Pull request merged**